executor/builder.go

```diff
@@ -1895,6 +1895,9 @@ func (builder *dataReaderBuilder) buildTableReaderForIndexJoin(ctx context.Conte
 	}
 	handles := make([]int64, 0, len(datums))
 	for _, datum := range datums {
+		if datum[0].IsNull() {
```

Comment:
Leave a `TODO` here: `After we propagate not null filter on join keys, this check can be removed.

Reply:
I prefer to use a github issue to trace this progress.

Reply:
@XuHuaiyu After rethinking, maybe it's better to:

In function `buildExecutorForIndexJoin()`, filter out `datums` which contains any `NULL` value. Use the `datums` that doesn't contains any NULL value to build the index key range of the inner table.